browser/ui/views/tabs/brave_tab_container.cc

```diff
@@ -444,22 +463,185 @@ BrowserRootView::DropIndex BraveTabContainer::GetDropIndex(
+    // tab.
+    center = is_tab_pinned ? tab->x() : tab->y();
+    const int length = is_tab_pinned ? tab->width() : tab->height();
+    if (drop_index < GetTabCount()) {
+      center += drop_before ? -(tabs::kVerticalTabsSpacing / 2) : (length / 2);
+    } else {
+      center += length + (tabs::kVerticalTabsSpacing / 2);
+    }
+  } else {
+    // Dropping before a group header.
+    TabGroupHeader* const header = group_views_[tab->group().value()]->header();
+    // Since there is no tab group in pinned tabs, there is no need to consider
+    // the x-axis.
+    center = header->y() + tabs::kVerticalTabsSpacing / 2;
+  }
+
+  // Determine the screen bounds.
+  gfx::Point drop_loc(is_tab_pinned ? center - g_drop_indicator_width / 2 : 0,
+                      is_tab_pinned ? tab->y() - g_drop_indicator_height
+                                    : center - g_drop_indicator_height / 2);
+  ConvertPointToScreen(this, &drop_loc);
+  gfx::Rect drop_bounds(drop_loc.x(), drop_loc.y(), g_drop_indicator_width,
+                        g_drop_indicator_height);
+
+  // If the rect doesn't fit on the monitor, push the arrow to the bottom.
+  display::Screen* screen = display::Screen::GetScreen();
+  display::Display display = screen->GetDisplayMatching(drop_bounds);
+  *is_beneath = !display.bounds().Contains(drop_bounds);
+
+  if (*is_beneath) {
+    drop_bounds.Offset(
+        is_tab_pinned ? 0 : drop_bounds.width() + tab->width(),
+        is_tab_pinned ? drop_bounds.height() + tab->height() : 0);
+  }
+
+  return drop_bounds;
+}
+
+gfx::ImageSkia* BraveTabContainer::GetDropArrowImage(
+    BraveTabContainer::DropArrow::Position pos,
+    bool beneath) {
+  return ui::ResourceBundle::GetSharedInstance().GetImageSkiaNamed(
+      pos == BraveTabContainer::DropArrow::Position::Vertical
+          ? (beneath ? IDR_TAB_DROP_UP : IDR_TAB_DROP_DOWN)
+          : (beneath ? IDR_TAB_DROP_LEFT : IDR_TAB_DROP_RIGHT));
+}
+
+void BraveTabContainer::SetDropArrow(
+    const absl::optional<BrowserRootView::DropIndex>& index) {
+  if (!index) {
+    controller_->OnDropIndexUpdate(absl::nullopt, false);
+    drop_arrow_.reset();
+    return;
+  }
+
+  // Let the controller know of the index update.
+  controller_->OnDropIndexUpdate(index->value, index->drop_before);
+
+  if (drop_arrow_ && (index == drop_arrow_->index())) {
+    return;
+  }
+
+  bool is_beneath;
+  gfx::Rect drop_bounds = GetDropBounds(index->value, index->drop_before,
+                                        index->drop_in_group, &is_beneath);
+
+  if (!drop_arrow_) {
+    DropArrow::Position position = DropArrow::Position::Vertical;
+    if (GetTabCount() > 0) {
+      Tab* tab = GetTabAtModelIndex(0);
+      position = tab->data().pinned ? DropArrow::Position::Vertical
+                                    : DropArrow::Position::Horizontal;
+    }
+    drop_arrow_ =
+        std::make_unique<DropArrow>(*index, position, is_beneath, GetWidget());
```

Comment:
Owning a view is not recommended, as widget usually owns them. 

I think we should have either weak_ptr<DropArrow>. And then instead of reset(), we can call drop_arrow_->GetWidget()->Close().

Reply:
`drop_arrow_` does not possess a GetWidget() method, given that it is an instance of the WidgetObserver class.

Reply:
My bad. I thought DropArrow is a view. Thanks for the correction.